go-controller/pkg/cni/ovs.go

```diff
@@ -98,17 +99,54 @@ func ofctlExec(args ...string) (string, error) {
 	return stdoutStr, nil
 }
 
-func waitForPodFlows(mac string) error {
+func waitForPodFlows(mac string, ipNets []*net.IPNet) error {
+	// query represents the match criteria, and different OF tables that this query may match on
+	type query struct {
+		match  string
+		tables []int
+	}
+
 	return wait.PollImmediate(200*time.Millisecond, 20*time.Second, func() (bool, error) {
-		// Query the flows by mac address
-		query := fmt.Sprintf("table=9,dl_src=%s", mac)
-		// ovs-ofctl dumps error on stderr, so stdout will only dump flow data if matches the query.
-		stdout, err := ofctlExec("dump-flows", "br-int", query)
-		if err != nil {
-			return false, nil
+		// Function checks for OpenFlow flows to know the pod is ready
+		// TODO(trozet): in the future use a more stable mechanism provided by OVN:
+		// https://bugzilla.redhat.com/show_bug.cgi?id=1839102
+
+		// Query the flows by mac address for in_port_security
+		queries := []query{
+			{
+				match:  "dl_src=" + mac,
+				tables: []int{9},
+			},
+		}
+		for _, ipNet := range ipNets {
+			var ipMatch string
+			if ipNet.IP.To4() != nil {
```

Comment:
`!utilnet.IsIPv6(ip)` is preferred to `ip.To4() != nil`

Reply:
done